### PR TITLE
ssh-import-id: refactor and move to pkgs/by-name

### PR DESCRIPTION
**Pull request by fabaff**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 350322` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>ssh-import-id</li>
    <li>ssh-import-id.dist</li>
  </ul>
</details>

**Review by viraptor — APPROVED** [review]

File: pkgs/by-name/ss/ssh-import-id/package.nix [hunk 1]
```diff
@@ -0,0 +1,63 @@
+{
+  lib,
+  extraHandlers ? [ ],
+  fetchgit,
+  installShellFiles,
+  makeWrapper,
+  python3,
+}:
+
+python3.pkgs.buildPythonApplication rec {
```
Comment (mkg20001) [inline pkgs/by-name/ss/ssh-import-id/package.nix:10]: This shouldn use python3.pkgs

Also is buildPythonApplication right? Why not buildPythonPackage?

File: pkgs/by-name/ss/ssh-import-id/package.nix [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+  build-system = with python3.pkgs; [ setuptools ];
+
+  nativeBuildInputs = [
+    makeWrapper
+    installShellFiles
+  ];
+
+  dependencies =
+    with python3.pkgs;
```
Comment (mkg20001) [inline pkgs/by-name/ss/ssh-import-id/package.nix:34]: Neither should this use python3.pkgs

File: pkgs/by-name/ss/ssh-import-id/package.nix (hunk 1 above)
Comment (fabaff) [reply pkgs/by-name/ss/ssh-import-id/package.nix]: > This shouldn use python3.pkgs

I guess that the slicing issue is back as the manual indeed says to use `python3Packages`. Fixed.

> Also is buildPythonApplication right? Why not buildPythonPackage?

`buildPythonPackage` is for Python module. This is an application which is not imported by other modules.

File: pkgs/by-name/ss/ssh-import-id/package.nix [hunk 3]
```diff
@@ -0,0 +1,63 @@
+{
+  lib,
+  extraHandlers ? [ ],
+  fetchgit,
+  installShellFiles,
+  makeWrapper,
+  python3Packages,
+}:
+
+python3Packages.buildPythonApplication rec {
```
Comment (mkg20001) [inline pkgs/by-name/ss/ssh-import-id/package.nix:10]: Dont put python3Packages here, see https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#buildpythonapplication-function-buildpythonapplication-function for proper usage.

Comment (fabaff) [reply pkgs/by-name/ss/ssh-import-id/package.nix:10]: Can you be more specify what you think is wrong here? 

https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#buildpythonapplication-function-buildpythonapplication-function says...

```bash
{ lib
, python3Packages
, fetchPypi
}:

python3Packages.buildPythonApplication rec {
  pname = "luigi";
  version = "2.7.9";
 [...]
 ```

Comment (mkg20001) [reply pkgs/by-name/ss/ssh-import-id/package.nix:10]: Wait, i confused things. I was reviewing as if this was a python-module but this is outside of pkgs/development/python-modules so buildPythonApplication should be fine. my bad.

**Review by mkg20001 — APPROVED** [review]

**Pull request merged**